Cloning record changes and mobID

Aranclanos opened the pull request:
### Purpose
Adds a serverside mobID.
The cloning console will now create or update the records based on that ID.
Cloning records will now save the player's name, mind and characterSettings.
Renames the variable from cloning records ToxingDmg to ToxinDmg.


### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- [X]  I correctly commented my code
- [X]  My code is indented with tabs and not spaces
- [X]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

Aranclanos committed:
Adds a serverside mobID.
The cloning console will now create or update the records based on that ID.
Cloning records will now save the player's name, mind and characterSettings.
Renames the variable from cloning records ToxingDmg to ToxinDmg.

fomalsd commented on UnityProject/Assets/Scripts/Health/Living/LivingHealthBehaviour.cs:
```diff
@@ -10,8 +10,13 @@
+	/// <summary>
+	/// Server side, each mob has a different one and never it never changes
+	/// </summary>
+	public int mobID;
```
well, if it never changes then why make it publically editable?
```suggestion
	public int MobID => mobID;
	private int mobID;
```

fomalsd commented on UnityProject/Assets/Scripts/Medical/CloningConsole.cs:
```diff
@@ -24,32 +23,45 @@ public void ToggleLock()
-			var toxinDmg = 0;
-			var bruteDmg = mob.GetTotalBruteDamage();
-			CreateRecord(name, oxyDmg, burnDmg, toxinDmg, bruteDmg, uniqueIdentifier);
+			CreateRecord(name, oxyDmg, burnDmg, toxinDmg, bruteDmg, uniqueIdentifier, characterSettings, mobID, mind);
```
might've introduced a local struct to contain and transfer all that info in
instead of 9000 parameters

fomalsd commented on UnityProject/Assets/Scripts/Medical/CloningConsole.cs:
```diff
@@ -58,18 +70,29 @@ public class CloningRecord
+	public CloningRecord()
 	{
-		Name = name;
+		int scanID = Random.Range(0, 1000);
```
there's a really high chance of same id being generated, what happens if two records have the same id?
maybe just start from 0001 and keep incrementing it instead?

Aranclanos committed:
Changes mobID to be a private set only.
Lowers the excessive amount of arguments used to create new cloning records.
Changes the scan ID random chance from 1000 numbers to 10000.
